nixos/modules/services/video/epgstation.nix

```diff
@@ -0,0 +1,335 @@
+        <link xlink:href="https://github.com/l3tnun/EPGStation/blob/master/config/config.sample.json"/>
+      '';
+    };
+
+    readOnlyOnce = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Don't reload configuration files at runtime.";
+    };
+
+    serverPort = mkOption {
+      type = types.port;
+      default = 20772;
+      description = ''
+        Port for EPGStation to listen on. Be careful because EPGStation listens
+        on a total of two ports: <literal>serverPort</literal> for regular HTTP
+        and <literal>serverPort + 1</literal> for websockets.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for the EPGStation web interface.
+
+        <warning>
+          <para>
+            It is generally advised against exposing EPGStation to the open
+            internet. Only use it inside a trusted local network, or consider
+            putting it behind a VPN if you want remote access.
+          </para>
+        </warning>
+      '';
+    };
+
+    basicAuth = {
+      user = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "epgstation";
+        description = ''
+          Basic auth username for EPGStation. If <literal>null</literal>, basic
+          auth will be disabled.
+
+          <warning>
+            <para>
+              Basic authentication has known weaknesses, the most critical being
+              that it sends passwords over the network in clear text. Use this
+              feature to control access to EPGStation within your family and
+              friends, but dont't rely on it for security.
+            </para>
+          </warning>
+        '';
+      };
+
+      password = mkOption {
+        type = types.str;
+        default = "INSECURE_GO_CHECK_CONFIGURATION_NIX";
+        example = "password";
+        description = ''
+          Basic auth password for EPGStation. The password doesn't need to be
+          set if either basic auth is disabled or
+          <option>basicAuth.passwordFile</option> has been set.
+
+          <warning>
+            <para>
+              This stores the plaintext password in the world-readable Nix
+              store.  Consider using <option>basicAuth.passwordFile</option>
+              instead.
+            </para>
+          </warning>
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = types.path;
+        default = pkgs.writeText "epgstation-password" cfg.basicAuth.password;
+        example = "/run/keys/epgstation-password";
+        description = ''
+          A file containing the password corresponding to
+          <option>basicAuth.user</option>.
+        '';
+      };
+    };
+
+    mirakurunPath =  mkOption (let
+      # Should be kept in sync with services.mirakurun.serverSettings.path
+      sockPath = "/var/run/mirakurun/mirakurun.sock";
```

Comment:
Fixed. Since `config.services.mirakurun.serverSettings` is a YAML type, the `path` setting wasn't exposed. So I created an alias option `config.services.unixSocket`. I suppose it shouldn't be possible to create conflicting settings though, because I haven't used `mkDefault` for `serverSettings.path`. I didn't use freeform modules for this option because `config.services.mirkaurun.serverSettings.path` is a bit too obscure, especially when referenced from other modules.